### DISCUSSION TITLE
Using existing editor model content

### DISCUSSION THREAD

**Discussion by Ehesp** [0xTheProDev/Operational-Transformation#105] [General]
Hey,

I'm using `firebase-monaco` and it's working great, however when I connect the `FireMonacoEditor` class to my editor, I believe it attempts to read the history from RTDB and updates the model. If there is no history, the editor is essentially updated to show no content (an empty model).

The part I'm struggling to work out, is how do I apply existing content, so when the class connects it keeps the existing model content, and then applies history on-top of that? Is there some way I can essentially grab the current model, and apply that as a history item on the database?

Cheers for the great work on this.

**Comment by Ehesp** [comment]
Well, this seems to work - not sure if it's the best way though.

```js
fireMonaco.on(EditorClientEvent.Ready, () => {
  fireMonaco.text = < update with content
});
```

**Reply by 0xTheProDev** [reply]
Yes, this is the correct way. Similarily, you can take a stash before disposing the editor to persist content even when `FireMonaco` is no longer connected.
